mlir/include/mlir-c/Rewrite.h

```diff
@@ -301,6 +301,10 @@ mlirFreezeRewritePattern(MlirRewritePatternSet op);
 MLIR_CAPI_EXPORTED void
 mlirFrozenRewritePatternSetDestroy(MlirFrozenRewritePatternSet op);
 
+MLIR_CAPI_EXPORTED MlirLogicalResult mlirApplyPatternsAndFoldGreedilyForOp(
+    MlirOperation op, MlirFrozenRewritePatternSet patterns,
+    MlirGreedyRewriteDriverConfig);
```

Comment:
Not sure which name is suitable.. So the current name is just a placeholder. And modifying `mlirApplyPatternsAndFoldGreedily` seems a breaking change.

Reply:
> Can you please extract this change in an independent PR? This is largely unrelated (your test python pass could just do another kind of rewrite right now).

Ahh sure.